OpenSim/Sandbox/MatlabScripts/timesSeriesTable.m

```diff
@@ -0,0 +1,195 @@
+% Ajay Seth, Ayman Habib, Jen Hicks. 
+
+import org.opensim.modeling.*
+
+if strcmp( class(generictable), 'org.opensim.modeling.TimeSeriesTableVec3') ...
+   || strcmp( class(generictable), 'org.opensim.modeling.TimeSeriesTable') 
+   
+    output_table = table2tsc(generictable);
+
+elseif  strcmp( class(generictable), 'tscollection') 
+   
+    output_table = tsc2table(generictable);
+    
+else
+    error('input must either be an OpenSim TimeSeriesTable or a Matlab tscollection')
+end
+
+end
+
+
+function tsc = table2tsc(generictable) 
+    % convert a OpenSim timeSeriesTable to a Matlab timeseries object
+    import org.opensim.modeling.*
+
+    %% get the number or rows and columns
+    nCol = generictable.getNumColumns;
+    nRow = generictable.getNumRows;
+
+    %% instantiate an empty table
+    tsc = tscollection();
+
+    %% determine temp storage array size. 
+    if strcmp( char(generictable.getClass), 'class org.opensim.modeling.TimeSeriesTableVec3')
+        rowdata = zeros(nRow, 3);
+    elseif strcmp( char(generictable.getClass), 'class org.opensim.modeling.TimeSeriesTable')
+         % if data is double
+        rowdata = zeros(nRow, 1);
+    else
+        error('unkown class type: must either be a TimeSeriesTable or TimeSeriesTableVec3')
+    end
+
+    % get the time column. 
+    for iRow = 0 : nRow - 1
+        time(iRow+1,1) = generictable.getIndependentColumn.get(iRow);
+    end 
+
+
+    for iCol = 0 : nCol -1
+
+        % get the data column name
+        col_label  = char(generictable.getColumnLabels.get(iCol));
+
+        if strcmp( char(generictable.getClass), 'class org.opensim.modeling.TimeSeriesTableVec3');
+
+            for iRow = 0 : nRow - 1
+                rowdata(iRow+1,1) = generictable.getDependentColumnAtIndex(iCol).getElt(iRow,1).get(0);
+                rowdata(iRow+1,2) = generictable.getDependentColumnAtIndex(iCol).getElt(iRow,1).get(1);
+                rowdata(iRow+1,3) = generictable.getDependentColumnAtIndex(iCol).getElt(iRow,1).get(2);       
+            end
+
+        else strcmp( char(generictable.getClass), 'class org.opensim.modeling.TimeSeriesTable');
+
+            for iRow = 0 : nRow - 1
+                rowdata(iRow+1,1) = generictable.getDependentColumnAtIndex(iCol).getElt(iRow,0);
+            end
+        end
+
+        % append column (as a table) to the existing table. 
+        tso = timeseries(rowdata,time,'Name',col_label);
+        tsc = addts(tsc, tso);
+
+    end
+
+end
+
+function timeseriestable = tsc2table(tsc)
+    % convert a Matlab timeseries object to a OpenSim timeSeriesTable 
+
+    import org.opensim.modeling.*
+
+    %% get the column names
+    labelnames = tsc.gettimeseriesnames;
+
+    %% data size
+    dataSize = tsc.size();
+    nRow = dataSize(1); nlabels = dataSize(2);
+    [m, nCol] = size(tsc.(labelnames{1}).Data);
+    
+    %% empty opensim table
+    if nCol == 1
+        table = DataTable();
+    elseif nCol == 3
+        table = DataTableVec3();
```

Comment:
I think these should be TimeSeriesTables, so that `tsc2table(table2tsc(T))` gives back something of the same type as T.